pulumi/python/automation/main.py

```diff
@@ -85,28 +96,45 @@ def usage():
     print(usage_text, file=sys.stdout)
 
 
-def provider_instance(provider_name: str) -> Provider:
-    """Dynamically instantiates an infrastructure provider
-    :param provider_name: name of infrastructure provider
-    :return: instance of infrastructure provider
-    """
-    module = importlib.import_module(name=f'providers.{provider_name}')
-    return module.INSTANCE
+def write_env(env_config, stack_name):
+    """Create a new environment file and write our stack to it"""
+    with open(env_config.filename, 'w') as f:
+        try:
+            print("PULUMI_STACK=" + stack_name, file=f)
+            msg = 'Environment configuration file not found. Creating new file at the path: %s'
+            RUNNER_LOG.error(msg, env_config.filename)
+        except (FileNotFoundError, PermissionError):
+            RUNNER_LOG.error("Unable to build configuration file")
+            sys.exit(2)
+
+
+def append_env(env_config, stack_name):
+    """Append our stack to the existing environment file"""
+    with open(env_config.config_path, 'a') as f:
+        try:
+            msg = 'Environment configuration file does not contain PULUMI_STACK, adding'
+            print("PULUMI_STACK=" + stack_name, file=f)
+            RUNNER_LOG.error(msg)
```

Comment:
Same comment as above.  The `print` syntax is non explicit so I'd suggest a comment.  Also the `error` log adds to the confusion making me wonder why it's an error case here when `except` below seems to handle an error.

Reply:
Updated.